nixos/modules/services/web-apps/lemmy.nix

```
@@ -102,14 +129,14 @@ in
           rate_limit.image = 6;
           rate_limit.image_per_second = 3600;
         } // {
-        database = mapAttrs (name: mkDefault) {
-          user = "lemmy";
-          host = "/run/postgresql";
-          port = 5432;
-          database = "lemmy";
-          pool_size = 5;
-        };
-      });
+          database = mapAttrs (name: mkDefault) {
+            user = "lemmy";
+            host = "/run/postgresql";
+            port = 5432;
+            database = "lemmy";
+            pool_size = 5;
+          };
+        }) // lib.foldlAttrs (acc: option: data: acc // lib.setAttrByPath data.setting { _secret = option; }) {} secrets;
```

I think this is more clear.
```suggestion
        }) // lib.foldlAttrs (acc: option: data: acc // lib.setAttrByPath data.setting { _secret = "$CREDENTIALS_DIRECTORY/${option}"; }) {} secrets;
```

---

I don't think that this is a good idea since we depend on the secret paths not being escaped, I added a comment to prevent confusion.